components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -364,20 +364,21 @@ private void preSetUserClaimsOnEmailUpdate(Map<String, String> claims, UserStore
         if (StringUtils.isNotBlank(emailAddress)) {
 
             String existingEmail;
-            String username = IdentityUtil.addDomainToName(user.getUserName(), user.getUserStoreDomain());
+            String username = user.getUserName();
```

Comment:
Don't we need to append userstore domain here?
does user.getUserName() gives name with userstore domain

Reply:
Appending the userstore name was the error.

user.getUserName() does not contain the userstore domain